ElunaLoader.cpp

```diff
@@ -196,7 +196,17 @@ void ElunaLoader::ReadFiles(lua_State* L, std::string path)
 bool ElunaLoader::CompileScript(lua_State* L, LuaScript& script)
 {
     // Attempt to load the file
-    int err = luaL_loadfile(L, script.filepath.c_str());
+    int err = 0;
+    if (script.fileext == ".moon")
+    {
+        std::ifstream t(script.filepath);
+        std::string str((std::istreambuf_iterator<char>(t)), std::istreambuf_iterator<char>());
+        t.close();
+
+        str = "return require('moonscript').loadstring([[" + str + "]])()";
```

Comment:
Isnt it wrong here to call the function at the end? We want to just load the script but not execute it.
So we would want to call loadstring only - not call the loaded string afterwards.
```suggestion
        str = "return require('moonscript').loadstring([[" + str + "]])";
```

Reply:
Ah, I see how it is.
The line below just does loadstring instead of actually running the code.

Do you think it would be possible instead to actually run the moonscript loadstring/loadfile here?

Reply:
In addition to the change suggested above, one would need to change the luaL_loadstring to luaL_dostring. So the complete code would be alon the lines of the following

```cpp
    int err = 0;
    if (script.fileext == ".moon")
    {
        std::string str = "return require('moonscript').loadfile([[" + script.filepath+ "]])";
        err = luaL_dostring(L, str.c_str());
    } else
        err = luaL_loadfile(L, script.filepath.c_str());

```